src/effects/deeplearning/DeepLearningEffectBase.cpp

```diff
@@ -33,17 +33,26 @@ DeepLearningEffectBase::DeepLearningEffectBase()
 
 bool DeepLearningEffectBase::Init()
 {
-   DeepModelManager &manager = DeepModelManager::Get();
+   // Catch any errors while setting up the DeepModelManager
+   try 
+   {
+      DeepModelManager &manager = DeepModelManager::Get();
```

Comment:
End try block here, e.g.:

``` cpp
try
{
    DeepModelManager &manager = DeepModelManager::Get();
}
catch (InvalidModelCardDocument &e)
{
...
}

// try loading the model (if available)
```

Reply:
discussed offline, cannot move "try loading the model..." code outside of the try block because of scope.